### PR TITLE
Speaker renaming with a persistent, searchable name directory

### PR DESCRIPTION
**Pull request by DanielSolomon**
## Summary

Implements manual speaker relabeling (the alternative discussed in #24): click a speaker label in any transcript to assign a real name.

- **Rename anywhere** — clicking a "Speaker A" / "דובר א׳" label (hover underlines it) opens a popover in both the recording detail view and the live transcript mid-recording, sharing one `SpeakerNamePicker` component.
- **Persistent directory** — names live in a new `SpeakerDirectory` (`speaker-names.json` in Application Support, app-wide like the tombstones, not the relocatable recordings folder). The popover is type-to-filter with an `Add "…"` row and a reset-to-default row; a "Known speakers" section in Settings → Speakers manages the list.
- **Names flow everywhere** — per-recording overlay map `Recording.speakerNames` (segments keep raw `SPEAKER_NN` IDs) resolves in the UI, Copy Transcript, the `.srt` sidecar (regenerated on rename), live SRT export, and the LLM title feed. Unnamed speakers fall back to the friendly label / raw ID exactly as before.
- **Live renames survive re-diarization** — the post-stop offline pass re-keys all speaker IDs; `SpeakerNameRemapper` carries names onto the new IDs via a duration-weighted vote over the index-aligned segment arrays (handles merges and splits). Batch re-transcribes clear names instead, since fresh clustering has no alignment to the old IDs.
- **Merge by same name** — naming two over-split IDs identically gives them a shared color and collapses their consecutive turns in copied/exported text.

## Test plan

- [x] 21 new unit tests: `SpeakerDirectoryTests` (persistence, dedupe, filtering), `SpeakerNameRemapperTests` (identity/swap/merge/split/edge cases), Recording codable round-trip + legacy decode, formatter/exporter with names incl. same-name collapse, `RecordingStore.setSpeakerName` persistence + SRT regeneration
- [x] Full suite: 452 unit tests pass (the 9 MilaUITests failures are the known XCUITest-foregrounding environment issue — CI runs those non-fatally)
- [x] Manual: built a signed 1.9.0-beta.1 DMG with bundled PythonRuntime and installed locally for end-to-end testing

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Assign custom display names to diarized speakers in both live sessions and recordings.
  * Copy live transcripts with speaker names and export SRT subtitles using assigned names.
  * Manage a persistent “Known speakers” list in Settings.
* **Improvements**
  * Enhanced adaptive microphone gain with better handling of quiet speech and sustained background noise.
  * Speaker labels now stay consistent through re-diarization and offline post-processing.
* **Bug Fixes**
  * Transcript previews and send-to-LLM formatting now respect assigned speaker names.
* **Documentation**
  * Updated local DMG build/signing guidance for “Mila Local Dev” and signature verification details.

### CODE REVIEW

File: Mila/App/MilaApp.swift [hunk 1]
```diff
@@ -249,6 +249,9 @@ struct MilaApp: App {
     @StateObject private var recordingSummarizer: RecordingSummarizer
     @StateObject private var voiceMemosSettings: VoiceMemosSettings
     @StateObject private var voiceMemosImporter: VoiceMemosImporter
+    /// Persistent, app-wide list of speaker names — the pick-list behind
+    /// the transcript's speaker rename popover.
+    @StateObject private var speakerDirectory = SpeakerDirectory()
```
[the remark this answers is not in the document]
Comment (DanielSolomon) [reply Mila/App/MilaApp.swift]: Fixed in 7527f21 — `speakerDirectory` is now declared without a default and assigned via `_speakerDirectory = StateObject(wrappedValue:)` inside `init()`, alongside the other app-wide objects.

File: Mila/Models/SpeakerDirectory.swift [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                .appendingPathComponent("Mila-UITest-Speakers-\(UUID())", isDirectory: true)
+            self.init(directory: tmp)
+            return
+        }
+        let appSupport = try! FileManager.default.url(for: .applicationSupportDirectory,
+                                                      in: .userDomainMask,
+                                                      appropriateFor: nil,
+                                                      create: true)
+        self.init(directory: appSupport.appendingPathComponent("Mila", isDirectory: true))
+    }
```
[the remark this answers is not in the document]
Comment (DanielSolomon) [reply Mila/Models/SpeakerDirectory.swift:46]: Fixed in 7527f21 — replaced the force-try with the non-throwing `urls(for:in:)` lookup and a temporary-directory fallback, preserving the `--ui-test-clean-store` path.

File: Mila/Views/SettingsView.swift [hunk 3]
```diff
@@ -1282,6 +1286,65 @@ private struct DiarizationSettingsTabContent: View {
...
+            }
+        }
+    }
+
+    private func addName() {
+        if directory.add(newName) != nil {
+            newName = ""
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (DanielSolomon) [reply Mila/Views/SettingsView.swift:1351]: Fixed in 7527f21 — added `speakers.known.row.<name>`, `speakers.known.remove.<name>`, `speakers.known.addField`, and `speakers.known.addButton` identifiers following the existing `speakers.*` convention.

**Comment by DanielSolomon** [comment]
Rebased onto `main` so the PR now contains only the two speaker-renaming commits — previously it was stacked on the (unmerged) #84/#85 branches and would have pulled their content in on merge.

One coordination note with #85: that PR adds a live copy-transcript button backed by a `clipboardText` property. On this branch that property doesn't exist, so whichever PR merges second should add `names: speakerNames` to `clipboardText`'s `TranscriptFormatter.plainText` call (one line) so mid-recording copies also use assigned speaker names.

**Review by urisland — APPROVED** [review]
Reviewed for merge into the current beta line. Verified:

- **Merges cleanly** into current `main` (which now has #87 + #91) — no rebase needed despite the overlapping files; the changes sit in different regions.
- **App-wide-settings convention** (CLAUDE.md): `speakerDirectory` is declared without an inline default, assigned via `_speakerDirectory = StateObject(wrappedValue:)` in `init()`, and injected with `.environmentObject()` into **both** the main window and the Settings scene. ✓
- **All 3 CodeRabbit findings addressed + confirmed** (init-time StateObject, force-try → non-throwing `urls(for:in:)`, a11y identifiers).
- **Clean data model**: names are a per-recording `speakerNames` overlay; segments keep raw `SPEAKER_NN` IDs. `SpeakerNameRemapper` carries names across the offline re-diarize via a duration-weighted vote; batch re-transcribes clear names (no alignment). 21 new unit tests + legacy-decode round-trip.

Approving. The voice-recognition sibling (#97) is deferred to the next beta and will rebase to layer on top of this.
